services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -1005,6 +986,49 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
         //TODO(andriidovhan) check error message when the issue is fixed
     }
 
+    @Tidy
+    def "Flow status accurately represents the actual state of the flow and flow rules"() {
+        when: "Create a flow on a long path"
+        def swPair = topologyHelper.switchPairs.first()
+        def longPath = swPair.paths.max { it.size() }
+        swPair.paths.findAll { it != longPath }.each { pathHelper.makePathMorePreferable(longPath, it) }
+        def flow = flowHelperV2.randomFlow(swPair)
+        northboundV2.addFlow(flow)
+
+        then: "Flow status is changed to UP only when all rules are actually installed"
+        northbound.getFlowStatus(flow.flowId).status == FlowState.IN_PROGRESS
+        Wrappers.wait(PATH_INSTALLATION_TIME) {
+            assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP
+        }
+        def flowInfo = database.getFlow(flow.flowId)
+        def flowCookies = [flowInfo.forwardPath.cookie.value, flowInfo.reversePath.cookie.value]
+        def switches = pathHelper.getInvolvedSwitches(flow.flowId)
+        withPool(switches.size()) {
+            switches.eachParallel { Switch sw ->
+                assert northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.containsAll(flowCookies)
+            }
+        }
+
+        when: "Delete flow"
+        def deleteResponse = northboundV2.deleteFlow(flow.flowId)
+
+        then: "Flow is actually removed from flows dump only after all rules are removed"
+        northbound.getFlowStatus(flow.flowId).status == FlowState.IN_PROGRESS
+        Wrappers.wait(RULES_DELETION_TIME) {
+            assert !northbound.getFlowStatus(flow.flowId)
+        }
+        withPool(switches.size()) {
+            switches.eachParallel { Switch sw ->
+                assert !northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.any { it in flowCookies }
```

Comment:
when switch in a multiTable mode then we will have more than 2 rules per a flow
I propose to use special filter here
```
northbound.getSwitchRules(sw.dpId).flowEntries.findAll { cookie ->
    Cookie.isIngressRulePassThrough(cookie) || !Cookie.isDefaultRule(cookie)
}
```

Reply:
check now